File: 06/Code/Python/assembler.py
from parser import *
from code_generator import CodeGenerator
from symbol_table import SymbolTable
import sys, os.path

def assemble(infile, outfile):
    p = Parser(infile)
    st = SymbolTable()
    out = open(outfile, 'w')
    
    # Insert code here
    
    out.close()

    
if __name__=='__main__':
    if len(sys.argv)!=2:
        print 'Usage: %s infile.asm'%sys.argv[0]
        exit(0)
    infile = sys.argv[1]
    outfile = os.path.splitext( infile )[0] + '.hack'
    assemble(infile, outfile)

File: 05/Code/Java/Assembler.java
class Assembler
{
    public static void assemble(String infile, String outfile)
    {
        Parser p = new Parser(infile);
        SymbolTable st = new SymbolTable();
        
        // Insert code here
    }

    public static void main(String[] args)
    {
        if(args.length!=1)
        {
            System.err.println("Usage: java Assembler infile.asm");
            System.exit(0);
        }
        
        String infile = args[0];
        int dot = infile.lastIndexOf('.');
        String outfile = ((dot == -1) ? infile : infile.substring(0, dot) ) + ".hack";
        assemble(infile, outfile);
    }
}
